### PR TITLE
BUG: Inconsistent behavior when converting strings to floats #9924

### PR DESCRIPTION
**Pull request by sidjee**
Fix for Issue #9924
Added else if to give inf output on out of bounds input.

### CODE REVIEW

**Comment by eric-wieser** [comment]
I'm not sure this looks right, but at the very least it needs a test and a comment explaining what that line is for.

While you're doing that, the commit message should be changed from `BUG` to something like

    BUG: Convert very large numeric strings to infinity

    Fixes #9924 

So that it auto-closes the issue, and starts with the right prefix (thanks for fixing the PR title, but that's not enough, unfortunately).

File: numpy/core/src/multiarray/numpyos.c [hunk 1]
```diff
@@ -574,6 +574,9 @@ NumPyOS_ascii_strtold(const char *s, char** endptr)
             *endptr = (char*)s;
         }
     }
+    else if(s){
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/numpyos.c]: Why are you checking `s`? We already dereference it on line 521, so you (and the compiler) knows it is not null.

**Comment by person142** [comment]
Note that underflow has the same issue:

```
>>> import numpy as np
>>> np.longdouble('1e-5000')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for long double: 1e-5000
```

**Comment by charris** [comment]
Note that in the future you should work in a branch, not master.

**Pull request closed without merging**